source/common/event/libevent_scheduler.cc

```diff
@@ -41,5 +43,49 @@ void LibeventScheduler::run(Dispatcher::RunType mode) {
 
 void LibeventScheduler::loopExit() { event_base_loopexit(libevent_.get(), nullptr); }
 
+void LibeventScheduler::initializeStats(DispatcherStats* stats) {
+  stats_ = stats;
+  evwatch_prepare_new(libevent_.get(), &onPrepare, this);
+  evwatch_check_new(libevent_.get(), &onCheck, this);
+}
+
+void LibeventScheduler::onPrepare(evwatch*, const evwatch_prepare_cb_info* info, void* arg) {
```

Comment:
These methods should not be members of `LibeventScheduler` (we never use `this`). Instead they can be standalone in this file in the anonymous namespace.

Reply:
We do access private fields of `self`, though.

Reply:
OK, good as is.